index 8f30cd68..77407812 100644
--- a/include/MNN/MNNDefine.h
+++ b/include/MNN/MNNDefine.h
@@ -35,8 +35,27 @@
 #define MNN_PRINT(format, ...) syslog(LOG_WARNING, format, ##__VA_ARGS__); fprintf(stderr, format, ##__VA_ARGS__)
 #define MNN_ERROR(format, ...) syslog(LOG_WARNING, format, ##__VA_ARGS__); fprintf(stderr, format, ##__VA_ARGS__)
 #else
-#define MNN_PRINT(format, ...) printf(format, ##__VA_ARGS__)
-#define MNN_ERROR(format, ...) printf(format, ##__VA_ARGS__)
+enum class Level {
+  Info = 0,
+  Error = 1,
+};
+extern "C" {
+void mnn_ffi_emit(const char *file, size_t line, Level level,
+                  const char *message);
+}
+#define MNN_PRINT(format, ...)                                                 \
+  {                                                                            \
+    char logtmp[4096];                                                         \
+    snprintf(logtmp, 4096, format, ##__VA_ARGS__);                             \
+    mnn_ffi_emit(__FILE__, __LINE__, Level::Info, logtmp);                     \
+  }
+
+#define MNN_ERROR(format, ...)                                                 \
+  {                                                                            \
+    char logtmp[4096];                                                         \
+    snprintf(logtmp, 4096, format, ##__VA_ARGS__);                             \
+    mnn_ffi_emit(__FILE__, __LINE__, Level::Error, logtmp);                    \
+  }
 #endif
 
 #ifdef DEBUG
